consensus/consensus-types/src/timeout_certificate.rs

```diff
@@ -14,18 +14,22 @@ use std::{collections::BTreeMap, fmt};
 #[derive(Deserialize, Serialize, Clone, Debug, PartialEq, Eq)]
 /// TimeoutCertificate is a proof that 2f+1 participants in epoch i
 /// have voted in round r and we can now move to round r+1.
+/// It also contains the highest quorum cert round from 2f+1 participants as well as the highest quorum cert.
+/// More details in safety rules.
+/// Signatures are signed on TimeoutForSigning structure (see timeout.rs).
 pub struct TimeoutCertificate {
     timeout: Timeout,
-    signatures: BTreeMap<Author, Ed25519Signature>,
+    signatures: BTreeMap<Author, (Round, Ed25519Signature)>,
```

Comment:
The round inside Timeout is the same round, right?

Reply:
the single Timeout only contains the highest qc round, this Round is the qc_round signed by 2f+1 nodes

Reply:
Got it. hqc_round is actually the highest hqc_round.

Reply:
yup, I probably can use <Timeout, Signature> for simplicity since Timeout doesn't contain QC anymore, before I was trying to save the 2f QC cost

Reply:
I think the way you did it is good. The names confused me, my bad.